### PR TITLE
Add prototype type and spacing scale

### PR DESCRIPTION
**Pull request by adamyeats**
This pull request adds a prototype typographic scale and spacing scale.

Screenshot 2019-08-16 at 21 45 26

## How were these values created?

- The width of the grid gutters is currently `1.5rem` (24px). 
- The base font size is `1rem` (16px).

We use *1.5* as our "magic number". The `line-height` of `body` is set to 1.5, and this value multiplied by the element's font size is `1.5rem`. The default `margin-bottom` on text is also `1.5rem`.

We spoke about the headings having a smaller margin and line height. We set the `margin-bottom` to `1.25rem`, and the `line-height` to 1.25. Using our major thirds scale, a heading with the font size of `3.052rem` (48.832px) will have a computed `line-height` of 61.04px.

For the vertical spacing scale, we set our "normal" spacing value to `1.5rem` (same as our gutter size and our text `line-height`). This is our "rhythm unit". Then, for every unit on the scale, we _multiply the previous number in the scale by 1.5_. For example, the next spacing unit up from `1.5rem` would be `2.25rem`.

TL;DR: the magic number is 1.5, unless you use a heading, in which case the number is 1.25.

This deviates a little from the values you defined [here](https://github.com/appsignal/design-language/pull/20#issuecomment-520821588), but using these values (to my eye) seemed to make the line height too squashed at smaller viewport sizes. Again, the values here are not important, you can substitute whatever "magic numbers" you choose.

## How to test

Ensure `design-language` is checked out to this branch.  Ensure `component-library` is checked out to `feature/topbar-component` to see these changes in the "Typography" story, and how the spacings look in the `TopBar`.

## Why not fluid type?

After experimenting with [fluid type](https://www.smashingmagazine.com/2016/05/fluid-typography/), I decided to not use it for this feature as, when I used the code examples in the link, the text didn't look right and vertical rhythm appeared to be extremely unpredictable. I do think, however, this may be potentially super useful for UI elements in the app, which are more independent of vertical rhythm.

### CODE REVIEW

**Comment by wesoudshoorn** [comment]
Hey @xadamy sorry I had a day with 100% calls!

The font-sizes and heading sizes look good. I've got some comments on the line-heights and margin-bottoms being the same on all font-sizes:

*Heading line-heights*
image**
I think the line-height for the paragraphs is good, on smaller screens as well. The line-height for the larger headings don't feel right yet. On huge font-sizes, the vertical space between lines needs to be narrower. The vertical space is somethings big enough to easily follow the lines, and keeping a 1.25 ratio might be too big. Here's the same headline with a 1.1 (I might even argue that this can even be smaller).

image

**Margin-bottoms**
Having a margin-bottom of 1.5rem does not seem to work on every headline / copy on our marketing pages. In a blog post, the following works great
image

But in some other layouts, you really want to use the proximity of elements to connect them, especially if you're using smaller headlines headings / copy such as this example:

image

Would you then create variations of these headings with less margin bottom?

**Responsiveness**
There's no responsiveness yet on these headlines. Did you have any thoughts on this, or should we implement that later?

**Comment by adamyeats** [comment]
@wesoudshoorn Ok, that's all fair enough, next steps for you are to make changes to this branch until it suits you and then we can go ahead and merge this 👍

**Comment by wesoudshoorn** [comment]
Good, will do!

**Comment by adamyeats** [comment]
@wesoudshoorn Apologies for not answering your last two questions!

> Would you then create variations of these headings with less margin bottom?

Yes, to me, these are marked up a little differently than a header/paragraph combo.

> There's no responsiveness yet on these headlines. Did you have any thoughts on this, or should we implement that later?

Until we have breakpoints, essentially we don't really have anything to implement here, so let's come back to sizing the type responsively once we have breakpoints.

**Comment by wesoudshoorn** [comment]
I don't have work on these improvements because I am focussing on getting the _current_ homepage setup ready for the brand launch.

I am merging this as is, this way we have something to work with. We can easily change it later with a small PR from my end.
